src/components/DataViewer/DataViewer.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import React from "react";
+import styled from "styled-components";
+import Modal from "react-modal";
+
+interface DataViewerProps {
+    data: { [key: string]: string | number | null };
+    title: string;
+    isOpen: boolean;
+    onRequestClose: () => void;
+}
+
+const ModalWithStyle = styled(Modal)`
+    position: absolute;
+    inset: 50% 40px 40px 50%;
+    border: 0px;
+    background: rgb(255, 255, 255);
+    overflow: auto;
+    border-radius: 5px;
+    outline: none;
+    padding: 0px;
+    width: 80%;
+    height: 80%;
+    box-shadow: rgb(200, 200, 200) 0px 2px 10px;
+    transform: translate(-50%, -50%);
+    overflow: hidden;
+`;
+
+const Header = styled.h3`
+    display: flex;
+    justify-content: space-between;
+    background-color: #eeeeee;
+    margin-bottom: 1rem;
+    margin-top: 0;
+    padding: 0.5rem;
+`;
+
+const Content = styled.div`
+    overflow: scroll;
+    height: calc(100% - 5rem);
```

Comment:
is there a cleaner way to do this? the modal can be smaller if there's less data to show

Reply:
The modal now is responsive to the page size and the amount of data to show. We cleaned it with display flex.